app/src/main/res/layout/activity_main.xml

```diff
@@ -8,85 +8,75 @@
+        tools:text="@string/app_name"
         />
 
-    <TextView
-        android:id="@+id/textView4"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="@string/keyword"
-        android:textSize="@dimen/sp_17"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintHorizontal_bias="0.048"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent"
-        app:layout_constraintVertical_bias="0.189"
-        tools:text="KeyWord"
-        />
-    <EditText
-        android:id="@+id/edtSearch"
-        android:layout_width="@dimen/dp_170"
-        android:layout_height="wrap_content"
-        android:hint="@string/keyword"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent"
-        app:layout_constraintVertical_bias="0.166"
-        tools:layout_editor_absoluteX="75dp"
-        />
-    <EditText
-        android:id="@+id/edtNumberLimit"
-        android:layout_width="@dimen/dp_170"
-        android:layout_height="wrap_content"
-        android:hint="@string/type_number"
-        android:inputType="number"
+    <Button
+        android:layout_width="100dp"
+        android:layout_height="60dp"
+        android:onClick="onSearchButtonClicked"
+        android:text="@string/search"
+        android:textAllCaps="false"
+        android:textSize="@dimen/dp_16"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintTop_toTopOf="parent"
-        app:layout_constraintVertical_bias="0.282"
+        app:layout_constraintVertical_bias="0.578"
+        tools:text="Submit"
+        style="@style/Widget.AppCompat.Button.Colored"
         />
 
-    <Button
-        android:layout_width="wrap_content"
+    <android.support.design.widget.TextInputLayout
+        android:id="@+id/txtInputLayoutNumberLimit"
+        android:layout_width="@dimen/dp_368"
         android:layout_height="wrap_content"
-        android:onClick="onSearchButtonClicked"
-        android:text="@string/search"
-        android:textAllCaps="false"
+        app:errorTextAppearance="@style/ErrorText"
+        app:hintTextAppearance="@style/LabelText"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintLeft_toLeftOf="parent"
         app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintTop_toTopOf="parent"
-        app:layout_constraintVertical_bias="0.429"
-        tools:text="Submit"
-        />
+        app:layout_constraintVertical_bias="0.35"
+        >
+
+        <EditText
+            android:id="@+id/edtNumberLimit"
```

Comment:
WARNING: [TextFields] The view name (`@+id/edtNumberLimit`) suggests this is a number, but it does not include a numeric `inputType` (such as '`numberSigned`')
 Providing an `inputType` attribute on a text field improves usability because depending on the data to be input, optimized keyboards can be shown to the user (such as just digits and parentheses for a phone number). Similarly,a hint attribute displays a hint to the user for what is expected in the text field.

The lint detector also looks at the `id` of the view, and if the id offers a hint of the purpose of the field (for example, the `id` contains the phrase `phone` or `email`), then lint will also ensure that the `inputType` contains the corresponding type attributes.

If you really want to keep the text field generic, you can suppress this warning by setting `inputType="text"`.